src/redisearch_rs/trie_rs/src/node/mod.rs

```diff
@@ -963,6 +963,56 @@ impl<Data> Node<Data> {
             _phantom: Default::default(),
         }
     }
+
+    /// Decompose the node into:
+    ///
+    /// - Its children pointers, that will be appended to the provided buffer in reverse order.
+    /// - Its payload, returned by the function.
+    ///
+    /// The heap allocation backing the node will be freed.
+    pub(crate) fn into_raw_parts_reversed(
+        mut self,
+        children_buffer: &mut Vec<Node<Data>>,
+    ) -> Option<Data> {
+        let data = self.data_mut().take();
+        let n_children = self.n_children() as usize;
+        let ptr = self.downgrade();
+
+        children_buffer.reserve(n_children);
+
+        // SAFETY:
+        // - The pointer is well-aligned and points immediately after
+        //   the last element.
+        //   It is therefore true that all memory between the base pointer
+        //   and the offsetted pointer belongs to a single allocation.
+        let mut next_child = unsafe { ptr.children().ptr().add(n_children) };
+        for _ in 0..n_children {
+            // SAFETY:
+            // - `i` is strictly smaller than `n_children`, so we're in bounds.
```

Comment:
No `i`
```suggestion
            // - iteration number is smaller than `n_children`, so we're in bounds.
```
```suggestion
            // - `_` is strictly smaller than `n_children`, so we're in bounds.
```

Reply:
Fixed in [518b221](https://github.com/RediSearch/RediSearch/pull/6129/commits/518b2217a7f749bf00049a8ab3a5e8a414ce5732)